### ISSUE TITLE
[Feature Request] Make web frontend an actual frontend

### ISSUE THREAD

**Issue by Xcelled** [RocketMap/RocketMap#1141]
Right now, the frontend is a glorified terminal screen. It's pretty dumb in terms of functionality; it can't really take any action on its own without constant supervision from the webserver.

Consider pokemon. Expired pokemon are only removed from the map when they don't show up in the results from the server, despite the client showing them as "expired". In the extreme cases, shutting down the server leads to a map with a ton of pokemon on them... all expired. There's no need for this - the client should really take care of hiding pokemon on its own.

Another example is the fact there's no 'cannot reach server' notification on the map, so if your internet (or the server) goes down, you might never know that something broke unless you open the console.

Bottom line: frontend should be an actual frontend.

**Comment by rrbailey89** [comment]
:+1

**Comment by DiscoTim** [comment]
i really like the idea of a status/connected in the front end, and very simple to do.

how much of an issue though is not having the server dictate client actions? the Pokemon could hide themselves, but what else is effected by this? its not an offline app, although for sure could be made to function a little better when a server is offline.

**Comment by Xcelled** [comment]
My point is that this project evolved so that the web frontend was tied very tightly to the backend, really designed to be run side-by-side with a console window.

Separating the two can lead to a better design more in line with an "actual" web app, which is how a lot of people seem to be using this. Thinking of them as separate can help to uncover missing/awkward features like the two I mentioned.

**Comment by DiscoTim** [comment]
I agree with that, i think :) to me the fronted feels more like a dev and data visualization tool than something that could be a lot more useful to the majority of users.

**Comment by rrbailey89** [comment]
Personally I would very much like to be able to start the server. Tied to a file full of accounts. And then from the web page.. potentially tied behind some admin page deploy each worker to spots and the application intelligently handle each from there.

**Comment by sebastienvercammen** [comment]
@Xcelled You're coming at this with the wrong point of view (and certainly with the wrong attitude, after reading some of your messages on the other PRs/issues).

> My point is that this project evolved so that the web frontend was tied very tightly to the backend, really designed to be run side-by-side with a console window.

The front-end was developed only as an interface to see live Pokémon. Your description of "front-end" refers to a web app/platform, which PokémonGo-Map is not (originally).

I can also understand your points of wanting to control the workers from within the webpage, but separation of control (admin/console/settings) and entertainment (front-end, live Pokémon) were a primary concern.

This needs proper discussion with arguments, none of this **"I'm right just because I am!"** thinking.

To address your point of "PRs are slow": those most involved with the project are all active daily and come together on Discord (there's 26 of us there right now). We all have lives, and this is open source, so if positive change was what you were aiming for, perhaps joining us first (it's public!) would have been a better idea. Just a thought.

**Comment by Xcelled** [comment]
@sebastienvercammen 

>The front-end was developed only as an interface to see live Pokémon. Your description of "front-end" refers to a web app/platform, which PokémonGo-Map is not (originally).

>I can also understand your points of wanting to control the workers from within the webpage, but separation of control (admin/console/settings) and entertainment (front-end, live Pokémon) were a primary concern.

I understood that the map was geared towards developers and power users, however I saw an opportunity for the project to evolve towards a larger audience - and so did many others. This issue was supposed to be a place for discussing such changes, but that happened in a couple of other places instead.

> "I'm right just because I am!" thinking

If I've ever come across this way, I apologize. It's not my intent. There's plenty of examples where I specifically dropped my idea in favor of a better one from someone else. I will, however, argue tenaciously on any  points I feel could be done better, regardless of who proposed them.

>To address your point of "PRs are slow"

When I made that comment several months ago, there was a core team that was very opaque and mysterious. It wasn't clear what they were doing or how to 'join them' or if it was even possible. Most pull requests never got a response - not even a 'Thanks for your submission' message. This is very discouraging to any open source contributor who's given their time to a project. After enough instances of this, where I felt that *everyone's* work was being largely ignored, I said something about it, although I'll admit not in the best way and largely out of frustration.

I'm glad to hear things are different now, though.

**Comment by sebastienvercammen** [comment]
> I understood that the map was geared towards developers and power users

To clarify, the project tries to add to the social aspect of Pokémon Go, especially for small groups of friends or a very small neighborhood. In practice, it's often someone with a minimum in technical (and googling) skills that will set one up for their friends.

That's where the separation of front-end for entertainment and back-end for control came from :)

> If I've ever come across this way, I apologize. It's not my intent.

No problem, I know what it's like to be someone who will argue tenaciously on points that can be improved. I'll blame this one on "it's the internet, you don't have live social feedback or intent".

> I'm glad to hear things are different now, though.

I'm still trying to go through old PRs/issues and keep new ones up-to-date at the same time, but we all have lives and the work never ends. Always welcome over at #feedback on Discord if you want more direct replies, or even join us in #pr.

**Comment by FrostTheFox** [comment]
Feature Requests are no longer are taken via GitHub. Please submit FRs using http://vote.devkat.org
